app/controllers/api/v1/customers_controller.rb

```diff
@@ -54,13 +68,28 @@ def customer
         @customer ||= Customer.find(params[:id])
       end
 
+      def set_display_customer_balance
+        val = params[:display_customer_balance]
+        return if val.nil?
+
+        unless val.in?(["true", "false", "1", "0"])
+          invalid_query_parameter(:display_customer_balance, :unprocessable_entity, "Not a boolean")
+        end
+        @display_customer_balance = ActiveModel::Type::Boolean.new.cast(val)
+      end
+
       def authorize_action
         authorize! action_name.to_sym, customer
       end
 
       def search_customers
         customers = visible_customers.includes(:bill_address, :ship_address)
         customers = customers.where(enterprise_id: params[:enterprise_id]) if params[:enterprise_id]
+
+        if @display_customer_balance
+          customers = CustomersWithBalance.new(customers: customers).query
+        end
```

Comment:
This is generating a complete new query based on customer ids. It resets the `includes(:bill_address, :ship_address)` and could have a negative impact on performance. Wouldn't it be better to take the existing query and add the customer balance part on top?

`CustomersWithBalance` could be greatly simplified by taking a customers relation in the initializer. Other places can then call it with `Customer.of(enterprise)`. Much better division of responsibilities.